src/states/node.rs

```diff
@@ -2127,8 +2143,17 @@ impl Node {
         //       flow for joining nodes is in place and we send the routing table to the new node
         //       at the point where it gets added to the section.
         let pfx_name = prefix.lower_bound();
-        // if !prefix.is_compatible(self.our_prefix()) {
-        while let Some(rt_pfx) = self.peer_mgr.routing_table().find_section_prefix(&pfx_name) {
+        if !prefix.is_compatible(self.our_prefix()) {
+            while let Some(rt_pfx) = self.peer_mgr.routing_table().find_section_prefix(&pfx_name) {
+                if rt_pfx.bit_count() >= prefix.bit_count() {
+                    break;
+                }
+                debug!("{:?} Splitting {:?} on section update.", self, rt_pfx);
+                let _ = self.handle_section_split(rt_pfx, rt_pfx.lower_bound(), outbox);
+            }
+        }
+
+        if let Some(rt_pfx) = self.peer_mgr.routing_table().find_section_prefix(&pfx_name) {
             if merge && rt_pfx.bit_count() > prefix.bit_count() {
```

Comment:
The previous code wouldn't do that for merges, only splits.  I didn't want to change the existing functionality, but we did notice this and wondered if it might have been an oversight.